battle.go

```diff
@@ -23,6 +24,12 @@ type Battle struct {
 	results    BattleResults
 }
 
+type BattleRule int
+
+const (
+	BattleRuleNoFaint BattleRule = 1 << iota
+)
+
```

Comment:
We should define the default battle rules here as well, similar to the pokemon validation rules.
(nit, kinda) Also, would prefer all rules to be `1` if something is enabled, and `0` if something is disabled, so `BattleRuleNoFaint` -> `BattleRuleFaint`.